nav2_behavior_tree/include/nav2_behavior_tree/bt_action_node.hpp

```diff
@@ -49,10 +49,15 @@ class BtActionNode : public BT::ActionNodeBase
     node_ = config().blackboard->template get<rclcpp::Node::SharedPtr>("node");
 
     // Get the required items from the blackboard
+    bt_loop_timeout_ =
+      config().blackboard->template get<std::chrono::milliseconds>("bt_loop_timeout");
     server_timeout_ =
       config().blackboard->template get<std::chrono::milliseconds>("server_timeout");
     getInput<std::chrono::milliseconds>("server_timeout", server_timeout_);
 
+    // Set spin_until_future_complete timeout
+    timeout_ = server_timeout_ < bt_loop_timeout_ ? server_timeout_ : bt_loop_timeout_;
```

Comment:
`timeout_` is never really used. `server_timeout_` is in most places. I'm kind of wondering why this is around. If you just used `bt_loop_timeout_` in place of `timeout_` in L326, I don't think anything changes and removes this variable

Reply:
> If you just used `bt_loop_timeout_` in place of `timeout_`

This could result in the total time that the BT node waits for exceeding the specified `server_timeout` value. Say if `bt_loop_timeout` is 10ms and `server_timeout` is 15ms, the node would wait for the server to timeout for 20ms.

Reply:
But you still do the subtraction to find if its larger, you just don't need this intermediary variable. `timeout_` is only used in this line https://github.com/ros-planning/navigation2/blob/5084f5617ae65c4973ced077f536263124d78d4e/nav2_behavior_tree/include/nav2_behavior_tree/bt_action_node.hpp#L326 which could be replaced as 

```
auto timeout = remaining > bt_loop_timeout_ ? bt_loop_timeout_ : remaining;
```

Reply:
Ah yes, you're right!